src/config.rs

```diff
@@ -17,21 +17,38 @@ pub const CONFIG_FILE: &str = "lazymc.toml";
 /// Configuration version user should be using, or warning will be shown.
 const CONFIG_VERSION: &str = "0.2.8";
 
-/// Load config from file, based on CLI arguments.
-///
-/// Quits with an error message on failure.
-pub fn load(matches: &ArgMatches) -> Config {
+pub fn load(matches: &ArgMatches) -> Vec<Config> {
     // Get config path, attempt to canonicalize
     let mut path = PathBuf::from(matches.get_one::<String>("config").unwrap());
     if let Ok(p) = path.canonicalize() {
         path = p;
     }
 
-    // Ensure configuration file exists
-    if !path.is_file() {
+    let paths: Vec<PathBuf> = if path.is_dir() {
+        path.read_dir()
+            .unwrap()
+            .filter_map(|entry| {
+                entry.ok().and_then(|entry| {
+                    let path = entry.path();
+                    if path.is_file() {
+                        Some(path)
+                    } else {
+                        None
+                    }
+                })
+            })
+            .collect()
+    } else if path.is_file() {
+        vec![path.clone()]
+    } else {
+        vec![]
+    };
```

Comment:
Is there a more elegant way to do this?

Reply:
This can be improved with some functional magic:

```diff
-        path.read_dir()
-            .unwrap()
-            .filter_map(|entry| {
-                entry.ok().and_then(|entry| {
-                    let path = entry.path();
-                    if path.is_file() {
-                        Some(path)
-                    } else {
-                        None
-                    }
-                })
-            })
-            .collect()
+        path.read_dir()
+            .unwrap()
+            .filter_map(|entry| entry.ok())
+            .map(|entry| entry.path())
+            .filter(|path| path.is_file())
+            .collect()
```

I think we can also use [walkdir](https://docs.rs/crate/walkdir/) here, but this is fine for now.

Reply:
Thank you, implemented